### PR TITLE
Added support for test phases logging

### PR DESCRIPTION
**Pull request by michaelk-stratoscale**

### CODE REVIEW

File: py/strato/racktest/infra/executioner.py [hunk 1]
```diff
@@ -218,82 +228,86 @@ def _setDetachedData(self):
...
-            self._test._clusters = self._setUpDetachedClusters()
-        try:
-            getattr(self._test, 'setUp', lambda: None)()
-        except:
-            logging.exception(
-                "Failed setting up test in '%(filename)s'", dict(filename=self._filename()))
-            self.e_type, self.e_value, self.e_traceback = sys.exc_info()
-            suite.outputExceptionStackTrace()
-            raise
+        with self.logPhaseDuration("setup"):
```
Comment (rom-stratoscale) [inline py/strato/racktest/infra/executioner.py]: interesting - thought more about logging from the start of the test to this stage. But what u did make more sense I guess

File: py/strato/racktest/infra/executioner.py [hunk 2]
```diff
@@ -50,6 +52,14 @@ def host(self, name):
     def hosts(self):
         return self._hosts
 
+    @contextlib.contextmanager
```
Comment (rom-stratoscale) [inline py/strato/racktest/infra/executioner.py]: god bless context manager

Comment (michaelk-stratoscale) [inline py/strato/racktest/infra/executioner.py]: כפרה אליו!

**Comment by michaelk-stratoscale** [comment]
Merged

**Pull request closed without merging**